src/streams/receive.rs

```diff
@@ -381,8 +381,41 @@ impl StreamRx {
 
         let is_new_packet = register.update(seq_no, now, header.timestamp, clock_rate.get());
 
+        // Get the previous time for comparison
         let previous_time = self.last_time.map(|t| t.numer());
-        let time_u32 = extend_u32(previous_time, header.timestamp);
+
+        // Calculate the extended timestamp
+        let mut time_u32 = extend_u32(previous_time, header.timestamp);
+
+        // If we're coming out of a paused state and the timestamp would go backwards,
+        // incrementally adjust until we find a value that moves time forward
+        if self.paused && previous_time.is_some() && time_u32 < previous_time.unwrap() {
+            // In 32-bit RTP timestamps, adding 2^31 (MAX/2) flips to the other half of timestamp space
+            // This forces extend_u32 to produce a value in the next cycle
+            let half_cycle = 1u32 << 31;
+            let mut adjusted_ts = header.timestamp;
+
+            // Try at most twice (should be sufficient to find a suitable timestamp)
+            for _ in 0..2 {
```

Comment:
Adding `1 << 31` twice just brings you back to the same value so this doesn't need to be a loop

Reply:
True